app/src/main/java/com/example/studentarena/fragments/ProfileFragment.java

```diff
@@ -147,11 +163,7 @@ private void queryPosts() {
         allPosts.clear();
         ParseQuery<Post> query = ParseQuery.getQuery(Post.class);
         // include data referred by user key
-        query.include(Post.KEY_USER);
-        // limit query to latest 20 items
-        query.setLimit(20);
-        query.whereEqualTo("user", ParseUser.getCurrentUser());
-        query.addDescendingOrder("createdAt");
+        query.include(Post.KEY_USER).setLimit(20).whereEqualTo("user", ParseUser.getCurrentUser()).addDescendingOrder("createdAt");
```

Comment:
same comment as above. you can streamline this even more :)

also when lines like this get long it's common to break them up into one method call per line like this:
```
ParseQuery<Post> query = ParseQuery.getQuery(Post.class)
    .include(Post.KEY_USER)
    .setLimit(20)
    // and so on
```
it makes it easier to see, one line at a time, what all the steps in your little pipeline are. basically, what you had before, except you don't have to type `query` so many times

Reply:
Looked into it and made edit. The query.findinBackground won't work with that approach as am using it and not defining it.